app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8,
+                 load_factor_threshold: float = 0.7) -> None:
+        self.capacity = capacity
+        self.load_factor_threshold = load_factor_threshold
+        self.table = [[] for _ in range(self.capacity)]
+        self.size = 0
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_key = hash(key)
+        if self.size / len(self.table) > self.load_factor_threshold:
+            self.resize()
+        index = hash_key % len(self.table)
+        for i, element in enumerate(self.table[index]):
+            if element[0] == key:
+                self.table[index][i] = (key, value, hash_key)
+                break
+        else:
+            self.table[index].append((key, value, hash_key))
+            self.size += 1
+
+    def __getitem__(self, key: Any) -> Any:
+        index = hash(key) % len(self.table)
+        for element in self.table[index]:
+            if element[0] == key:
```

Comment:
element[0], element[1], element[2] ... better to make some dataclass or smth like that for every cell/node that you will use. It will improve readability of your code